api/src/test/java/org/openmrs/module/fgh/mpi/MpiHttpClientTest.java

```diff
@@ -1,11 +1,184 @@
+		
+		assertEquals(openCRResponse.get(FIELD_ACTIVE), patientResponse.get(FIELD_ACTIVE));
+		assertEquals(openCRResponse.get(OPENMRS_UUID), patientUuid);
+	}
+	
+	@Test
+	public void getPatient_shouldRetrievePatientInSanteMpi() throws Exception {
+		final String patientUuid = "patient-uuid-sante-mpi";
+		String baseUrl = "https://localhost:8080";
+		Map<String, Object> patientResponse = new HashMap<>();
+		patientResponse.put(FIELD_ACTIVE, true);
+		patientResponse.put(OPENMRS_UUID, patientUuid);
+		
+		when(MpiContext.initIfNecessary()).thenReturn(mpiContext);
+		when(mpiContext.getAuthenticationType()).thenReturn(AUTHENTICATION_TYPE.OAUTH);
+		when(mockMpiHttpClient.getPatient(patientUuid)).thenReturn(patientResponse);
+		doNothing().when(mockMpiHttpClient).doAuthentication(any(String.class));
+		Map<String, Object> santeMPIResponse = mockMpiHttpClient.getPatient(patientUuid);
+		
+		assertEquals(santeMPIResponse.get(FIELD_ACTIVE), patientResponse.get(FIELD_ACTIVE));
+		assertEquals(santeMPIResponse.get(OPENMRS_UUID), patientUuid);
+		
+	}
+	
 	@Test
-	public void addTests() throws Exception {
+	public void submitBundle_shouldRetrieveAccessTokenAndSubmitBundle() throws Exception {
+		final String patientUuid = "patient-uuid";
+		when(mockMpiHttpClient.getPatient(patientUuid)).thenReturn(singletonMap(FIELD_ACTIVE, false));
+		
+		List<Object> response = new ArrayList<>();
+		
+		Map<String, Object> messageHeader = FhirUtils.generateMessageHeader();
+		List<Map<String, Object>> entryList = new ArrayList<>();
+		entryList.add(messageHeader);
+		Map<String, Object> data = new HashMap<>();
+		Map<String, Object> resourceData = new HashMap<>();
+		resourceData.put("resourceType", "Bundle");
+		resourceData.put("resource", resourceData);
+		data.put("entry", entryList);
+		
+		String bundleData = data.toString();
+		when(mockMpiHttpClient.submitBundle("/fhiir-url/test", bundleData, List.class)).thenReturn(response);
+		List<Object> submutedData = mockMpiHttpClient.submitBundle("/fhiir-url/test", bundleData, List.class);
+		assertNotNull(submutedData);
 	}
 	
+	@Test
+	public void submitPatient_shouldSubmitPatientAsOpenCr() throws Exception {
+		
+		final String patientUuid = "patient-uuid-opencr";
+		Map<String, Object> patientData = new HashMap<>();
+		patientData.put(FIELD_ACTIVE, true);
+		patientData.put(OPENMRS_UUID, patientUuid);
+		List<Map<String, Object>> mpiIdsResp = new ArrayList<>();
+		when(mockMpiHttpClient.submitRequest("/fhiir-url/test", patientData.toString(), List.class)).thenReturn(mpiIdsResp);
+		this.mockMpiHttpClient.submitPatient(patientData.toString());
+		when(mpiContext.getAuthenticationType()).thenReturn(AUTHENTICATION_TYPE.CERTIFICATE);
+		
+		doAnswer(invocation -> {
+			Map<String, Object> capturedPatientData = (Map<String, Object>) invocation.getArguments()[0];
+			
+			assertEquals(capturedPatientData.get(FIELD_ACTIVE), patientData.get(FIELD_ACTIVE));
```

Comment:
Please put assertions outside the invocation, use variables to capture what you need to test and then check if the variable matches the expected value outside afterwards

Reply:
Did you see this comment @daniel-chirinda ?